diff --git a/node_modules/grandiose/binding.gyp b/node_modules/grandiose/binding.gyp
index 91d6bab..ef864f3 100644
--- a/node_modules/grandiose/binding.gyp
+++ b/node_modules/grandiose/binding.gyp
@@ -58,14 +58,14 @@
                     "library_dirs": [ "<(ndi_dir)/lib/lnx-x64" ]
                 }
             } ],
-            [ "OS == 'mac' and target_arch == 'x64'", {
+            [ "OS == 'mac' and (target_arch == 'x64' or target_arch == 'arm64')", {
                 "copies": [ {
                     "destination":  "build/Release",
-                    "files":        [ "<(ndi_dir)/lib/mac-x64/libndi.dylib" ]
+                    "files": [ "<(ndi_dir)/lib/mac/libndi.dylib" ]
                 } ],
                 "link_settings": {
                     "libraries":    [ "-Wl,-rpath,@loader_path", "-lndi" ],
-                    "library_dirs": [ "<(ndi_dir)/lib/mac-x64" ]
+                    "library_dirs": [ "<(ndi_dir)/lib/mac" ]
                 }
             } ]
         ]
diff --git a/node_modules/grandiose/ndi.js b/node_modules/grandiose/ndi.js
index fc38fea..9000296 100644
--- a/node_modules/grandiose/ndi.js
+++ b/node_modules/grandiose/ndi.js
@@ -76,7 +76,7 @@ const tmp       = require("tmp")
     }
     else if (os.platform() === "darwin") {
         /*  download NDI SDK distribution  */
-        const url1 = "https://downloads.ndi.tv/SDK/NDI_SDK_Mac/Install_NDI_SDK_v5_macOS.pkg"
+        const url1 = "https://downloads.ndi.tv/SDK/NDI_SDK_Mac/Install_NDI_SDK_v5_Apple.pkg"
         console.log("-- dowloading NDI SDK distribution")
         const data1 = await got(url1, { responseType: "buffer" })
         const file1 = tmp.tmpNameSync()
@@ -95,9 +95,9 @@ const tmp       = require("tmp")
         console.log("-- assembling NDI SDK subset")
         shell.rm("-rf", "ndi")
         shell.mkdir("-p", "ndi/include")
-        shell.mkdir("-p", "ndi/lib/mac-x64")
-        shell.mv(path.join(dir1, "NDI SDK for macOS/include/*.h"), "ndi/include/")
-        shell.mv(path.join(dir1, "NDI SDK for macOS/lib/macOS/*.dylib"), "ndi/lib/mac-x64/")
+        shell.mkdir("-p", "ndi/lib/mac")
+        shell.mv(path.join(dir1, "NDI SDK for Apple/include/*.h"), "ndi/include/")
+        shell.mv(path.join(dir1, "NDI SDK for Apple/lib/macOS/*.dylib"), "ndi/lib/mac/")

         /*  remove temporary files  */
         console.log("-- removing temporary files")
